pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -156,9 +149,21 @@ autoPatchelf() {
 
     # Add all shared objects of the current output path to the start of
     # cachedDependencies so that it's choosen first in findDependency.
-    cachedDependencies+=(
-        $(find "$prefix" \! -type d \( -name '*.so' -o -name '*.so.*' \))
-    )
+    # This error is ignored, because $prefix is not defined in this script.
+    # shellcheck disable=SC2154
+    while IFS='' read -r -d $'\0' file; do
+        cachedDependencies+=("$file")
+    done < <(find "$prefix" \! -type d \( -name '*.so' -o -name '*.so.*' \) -print0)
+
+    # Also add all runtime dependencies
+    if [ -n "${runtimeDependencies:-}" ]; then
+        for dep in $runtimeDependencies; do
+            while IFS='' read -r -d $'\0' file; do
+                cachedDependencies+=("$file")
+            done < <(find "$dep/lib" \! -type d \( -name '*.so' -o -name '*.so.*' \) -print0)
```

Comment:
Recursively adding all the `.so` files from a particular runtime dependency is going to also add files like `$dep/lib/somename/plugins/foo.so` to the RPATH. So maybe it's generally a better idea to list shared object files explicitly.